eth2util/registration/registration.go

```diff
@@ -0,0 +1,92 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package registration
+
+import (
+	"encoding/hex"
+	"strings"
+	"time"
+
+	eth2v1 "github.com/attestantio/go-eth2-client/api/v1"
+	eth2p0 "github.com/attestantio/go-eth2-client/spec/phase0"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/z"
+	"github.com/obolnetwork/charon/eth2util"
+)
+
+// DOMAIN_APPLICATION_BUILDER. See https://github.com/ethereum/builder-specs/blob/7b269305e1e54f22ddb84b3da2f222e20adf6e4f/specs/bellatrix/builder.md#domain-types.
+var registrationDomainType = eth2p0.DomainType([4]byte{0x00, 0x00, 0x00, 0x01})
+
+// NewMessage returns a v1.ValidatorRegistration message with the provided pubkey, feeRecipient, gasLimit and timestamp.
+func NewMessage(pubkey eth2p0.BLSPubKey, feeRecipient string, gasLimit uint64, timestamp time.Time) (eth2v1.ValidatorRegistration, error) {
```

Comment:
eth2client types should always use pointers, since that is how that libraary uses them